### DISCUSSION TITLE
为什么运行的是 prod 配置文件而不是 stage 配置文件？"build:stage": "UMI_ENV=stage max build"

### DISCUSSION THREAD

**Discussion by xXAvoraXx** [umijs/umi#11636] [Q&A] [answered]
为什么运行的是 prod 配置文件而不是 stage 配置文件？

"build:stage": "UMI_ENV=stage max build"

yarn build:stage

- config
- - config.dev.ts
- - config.prod.ts
- - config.stage.ts
- - config.ts

**Comment by fz6m** [comment]
参考 [`UMI_ENV`](https://umijs.org/docs/guides/env-variables#umi_env) 的配置文件优先级顺序，`config.prod.ts` 优先级高于 `config.${UMI_ENV}.ts` 。

**Reply by xXAvoraXx** [reply]
我没有完全理解。 为什么当 UMI_ENV=stage 应该根据配置优先级运行时，prod 却在运行？

**Reply by fz6m** [reply, marked as answer]
默认有开发和构建环境，当 `umi dev` 时是 `config.dev.ts` ，当 `umi build` 时是 `config.prod.ts` ，当用户还希望自己新增更多环境设定时，才需要使用 `UMI_ENV` ，此时就可以使用 `config.${UMI_ENV}.ts` 和 `config.${dev | prod | test}.${UMI_ENV}.ts` ，以便在不同用户自定义的环境有不同的配置（如环境变量的不同），这些配置会根据优先级进行覆盖，详见 https://umijs.org/docs/guides/env-variables#umi_env

**Reply by xXAvoraXx** [reply]
我创建了一个名为config.prod.stage.ts的配置文件，并且在构建项目时它运行了。谢谢。

**Reply by komorebi-cqd** [reply]
看你演示终于看懂了，我就说线上为啥老是用config.prod.ts的配置，指定了的根本不用，原来要在prod后面再加个词。
